### ISSUE TITLE
Why i can't set model body?

### ISSUE THREAD

**Issue by q52888940** [rehlds/ReGameDLL_CS#989]
void CAK47::Spawn()
{
	Precache();

	m_iId = WEAPON_AK47;
	SET_MODEL(edict(), "models/w_ak47.mdl");

	pev->body = 3;

**Comment by dystopm** [comment]
May you want to add this to weaponboxes instead of pure weapon entities?

**Comment by q52888940** [comment]
I want the BODY of the model to be 3 at the moment the player throws it, or when the armory_entity entity spawns

**Comment by q52888940** [comment]
> May you want to add this to weaponboxes instead of pure weapon entities?

I want the BODY of the model to be 3 at the moment the player throws it, or when the armory_entity entity spawns

**Comment by dystopm** [comment]
Yes, but you are not catching weaponboxes or armoury entities spawning. Weapon entities are attaches to players and they are invisible, they only have a model attached to them for touching purposes

**Comment by q52888940** [comment]
> 是的，但你没有捕捉到武器箱或军械库实体的生成。武器实体附着在玩家身上，它们是不可见的，它们只有一个模型附着在它们身上以供触摸

Can you provide me with the relevant code? I don't know where to write it

**Comment by Nord1cWarr1or** [comment]
This is by design — weapon entities attached to players are invisible "virtual" items (their model only exists for touch/pickup purposes); the visible world model with the body/submodel lives on the **weaponbox** (or armoury_entity), so `pev->body` must be set on those entities when they spawn, not inside `CAK47::Spawn()` (as explained by dystopm in the thread). Closing as completed (architecture explained).
